monai/handlers/stats_handler.py

```diff
@@ -91,13 +99,19 @@ def __init__(
         self.output_transform = output_transform
         self.global_epoch_transform = global_epoch_transform
         self.state_attributes = state_attributes
-        self.logger = logging.getLogger(name)
-        self._name = name
-
         self.tag_name = tag_name
         self.key_var_format = key_var_format
-        if logger_handler is not None:
-            self.logger.addHandler(logger_handler)
+        self.logger = logging.getLogger(name)  # if `name` is None, will default to `engine.logger` when attached
+        if name is not None:
+            if logging.root.getEffectiveLevel() > logging.INFO:
```

Comment:
normally the application/user of this `StatsHandler` should prepare appropriate loggers/handlers from the outside and pass them (by either the logger's name or logger handler) to this class. This is to decouple the logic of logging configuration and the actual code of generating logging content. So I think this addition should be in a logger config in https://github.com/Project-MONAI/tutorials/blob/master/3d_classification/ignite/densenet_training_dict.py instead of hard-coded here. cc @ericspod

Reply:
Hi @wyli ,

Thanks for your review and suggestions.
We already have the similar logic in `DataStats` transform:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utility/array.py#L594
Because for 90% users of `DataStats` or `StatsHandler`, they expect to see the log on screen, and I already received this feedback several times from @holgerroth and our SAs when they missed to call `logging.basicConfig(stream=sys.stdout, level=logging.INFO)`.
So maybe it's better to add `stdout` handler for user experience if they use this API to print stats?

Thanks.

Reply:
Hi @wyli ,

Updated in the latest commit.
Could you please help take a look again?

Thanks in advance.